backendApi/src/main/java/com/virtuslab/gitmachete/backend/api/IGitMacheteRepositorySnapshot.java

```diff
@@ -0,0 +1,20 @@
+package com.virtuslab.gitmachete.backend.api;
+
+import io.vavr.collection.List;
+import io.vavr.control.Option;
+
+import com.virtuslab.branchlayout.api.IBranchLayout;
+
+public interface IGitMacheteRepositorySnapshot {
```

Comment:
Can we have a few words of description about what do we mean by snapshot and why do we reflect this? It is (only) intuitive for me but some clarification would be nice.